standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -508,6 +511,48 @@ public static void clearQuickStats(Map<String, String> params) {
     params.remove(StatsSetupConst.NUM_ERASURE_CODED_FILES);
   }
 
+  public static void updateTableStatsForCreateTable(Warehouse wh, Database db, Table tbl,
+      EnvironmentContext envContext, Configuration conf, Path tblPath, boolean newDir)
+      throws MetaException {
+    // If the created table is a view, skip generating the stats
+    if (MetaStoreUtils.isView(tbl)) {
+      return;
+    }
+    assert tblPath != null;
+    if (tbl.isSetDictionary() && tbl.getDictionary().getValues() != null) {
+      List<java.nio.ByteBuffer> values = tbl.getDictionary().getValues().
+          remove(StatsSetupConst.STATS_FOR_CREATE_TABLE);
+      java.nio.ByteBuffer buffer;
+      if (values != null && values.size() > 0 && (buffer = values.get(0)).hasArray()) {
+        String val = new String(buffer.array(), StandardCharsets.UTF_8);
+        if (StatsSetupConst.TRUE.equals(val)) {
+          try {
+            boolean isIcebergTable =
+                HiveMetaHook.ICEBERG.equalsIgnoreCase(tbl.getParameters().get(HiveMetaHook.TABLE_TYPE));
+            PathFilter pathFilter = isIcebergTable ?
+                path -> !"metadata".equals(path.getName()) : FileUtils.HIDDEN_FILES_PATH_FILTER;
```

Comment:
This part is too Iceberg specific. Is it possible to get this filter via storage handler api?

Reply:
Move this part to `Table#isIcebergTable` on the client,
https://github.com/apache/hive/pull/4228/files#diff-a88cd54666ea2466fd0c0f1323efecdc0be83a3fbf726d471f94080e31affc15